addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -21,15 +22,266 @@ const BASERESOURCE = "esper-pioneer-shield-study";
 // XPCOMUtils.defineLazyModuleGetter(this, "studyUtils", STUDYUTILSPATH);
 */
 
+// telemetry utils
+const { TelemetryController } = Cu.import("resource://gre/modules/TelemetryController.jsm", null);
+const { TelemetrySession } = Cu.import("resource://gre/modules/TelemetrySession.jsm", null);
+const { TelemetryEnvironment } = Cu.import("resource://gre/modules/TelemetryEnvironment.jsm", null);
+
+XPCOMUtils.defineLazyModuleGetter(this, "PlacesUtils",
+  "resource://gre/modules/PlacesUtils.jsm");
+
 /**
  * Note: Setting studyUtils in this constructor due to above comments
  */
 class StudyTelemetryCollector {
+
   constructor(studyUtils, variation) {
     this.variation = variation;
     this.studyUtils = studyUtils;
   }
 
-  start() {
+  async start() {
+
+    this.studyUtils.telemetry({ event: "esper-init" });
+
+    // Ensure that we collect telemetry payloads only after it is fully initiated
+    Promise.all([
+      TelemetryEnvironment.onInitialized(),
+      TelemetrySession.delayedInit(),
```

Comment:
Mh, I think this might be wrong: isn't `TelemetrySession.delayedInit()` triggering a new delayed initialization of that component? That's potentially harmful and perf-impacting: you would end up with the same observers registered twice (once by Firefox itself, once by this study). Depending on when this function is called by the study, you might even regress startup performance (Telemetry's delayed init is supposed to start/finish 60 seconds after Firefox starts).

I think that all you really want/need here is to wait for Telemetry to completely initialize: you should probably just wait on `TelemetryController.promiseInitialized()` [link](http://searchfox.org/mozilla-central/rev/423b2522c48e1d654e30ffc337164d677f934ec3/toolkit/components/telemetry/TelemetryController.jsm#295).

Reply:
Yes, thank you! I'll add this to the shield docs. TelemetryEnvironment.onInitialized() and TelemetrySession.delayedInit() are false friends, TelemetryController.promiseInitialized() is the proper one :)

Reply:
Hmm, wait a minute, literally :) It takes forever (about a minute...) to wait for TelemetryEnvironment.onInitialized() - @Dexterp37 is there a way to trigger the telemetry initiation faster so that the time between code change and observing the results can become shorter?

Reply:
Hah, good one :-D It should take exactly 60 seconds for the delayed init to happen, so that's expected. Is this code meant to be used in the study or only in tests? If it's part of the study then no, we cannot make it happen earlier as it would impact startup performance. If it's only a matter of tests, then we can try to work around this limitation by forcing the "test mode".

Reply:
Test mode sounds great, tell me more! It'd be used only during certain tests and for development purposes to get a lower latency between code changes and the results :)

Reply:
Basically, just when testing, you can change the value of the `toolkit.telemetry.initDelay` preference (see [here](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/internals/preferences.html#testing)). A value of 1 might be good enough :) You need to chamge that before Firefox starts

Reply:
Perfect, thanks, works wonders! :)

Reply:
@Dexterp37 I was too quick too celebrate, [caused issues with some telemetry fields](https://github.com/motin/esper-pioneer-shield-study/pull/2/files#diff-ff2899c03c9e8e31a9b69d7621e08c83R37) - not important though for this study (it's fine to have the tests take 60 seconds)